openwisp_radius/api/views.py

```diff
@@ -106,11 +107,32 @@ def _get_ip_list(self, uuid):
     def _check_client_ip_and_return(self, request, uuid):
         client_ip, _is_routable = get_client_ip(request)
         ip_list = self._get_ip_list(uuid)
-        if bool(
-            (ip_list and client_ip in ip_list)
-            or (not ip_list and client_ip in app_settings.FREERADIUS_ALLOWED_HOSTS)
-        ):
-            return (AnonymousUser(), uuid)
+
+        if ip_list:
+            for ip in ip_list:
+                try:
+                    if ipaddress.ip_address(client_ip) in ipaddress.ip_network(ip):
+                        return (AnonymousUser(), uuid)
+                except ValueError:
+                    invalid_addr_message = _(
+                        f'Request rejected: ({ip}) in organization settings '
+                        'is invalid ip address. Please contact administrator.'
+                    )
+                    logger.warning(invalid_addr_message)
+                    raise AuthenticationFailed(invalid_addr_message)
+        else:
+            for ip in app_settings.FREERADIUS_ALLOWED_HOSTS:
+                try:
+                    if ipaddress.ip_address(client_ip) in ipaddress.ip_network(ip):
+                        return (AnonymousUser(), uuid)
+                except ValueError:
+                    invalid_addr_message = _(
+                        f'Request rejected: ({ip}) in application settings.py'
+                        'is invalid ip address. Please contact administrator.'
+                    )
+                    logger.warning(invalid_addr_message)
+                    raise AuthenticationFailed(invalid_addr_message)
```

Comment:
the two blocks of code look identical, let's avoid this.
Can you change `_get_ip_list` to fall back to `app_settings.FREERADIUS_ALLOWED_HOSTS` if None?